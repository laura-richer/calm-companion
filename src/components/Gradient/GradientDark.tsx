import React from 'react';
import LinearGradient from 'react-native-linear-gradient';
import { GradientProps } from './types';
import { colors } from '~styles/variables';

const GradientDark = ({ children, style, disabled }: GradientProps) => {
  return (
    <LinearGradient
      colors={disabled ? [colors.WHITE, colors.WHITE] : [colors.TWO, colors.THREE]}
      useAngle={true}
      angle={170}
      angleCenter={{ x: 0.75, y: 0.75 }}
      style={style}
    >
      {children}
    </LinearGradient>
  );
};

export default GradientDark;
