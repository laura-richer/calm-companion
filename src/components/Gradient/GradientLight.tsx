import React from 'react';
import LinearGradient from 'react-native-linear-gradient';
import { GradientProps } from './types';
import { colors } from '~styles/variables';

const GradientLight = ({ children, style }: GradientProps) => {
  return (
    <LinearGradient
      colors={[colors.FOUR, colors.FIVE]}
      useAngle={true}
      angle={145}
      angleCenter={{ x: 0.75, y: 0.75 }}
      style={style}
    >
      {children}
    </LinearGradient>
  );
};

export default GradientLight;
