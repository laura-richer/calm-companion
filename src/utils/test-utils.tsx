import React, { PropsWithChildren } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, RenderOptions } from '@testing-library/react-native';
import { Provider as StoreProvider } from 'react-redux';
import { EnhancedStore, configureStore } from '@reduxjs/toolkit';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { RootState } from '~store/index';
import navigationReducer, {
  initialState as navigationInitialState,
} from '~store/slices/navigation';
import quickHelpReducer, { initialState as quickHelpInitialState } from '~store/slices/quickHelp';

type DeepPartial<T> = {
  [K in keyof T]?: T[K] extends object ? DeepPartial<T[K]> : T[K];
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: EnhancedStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { preloadedState = {}, store = createTestStore(preloadedState) }: ExtendedRenderOptions = {}
) => {
  const wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </StoreProvider>
    );
  };

  return {
    ...render(ui, {
      wrapper,
    }),
  };
};

const _createStore = (preloadedState: Partial<RootState> = {}): EnhancedStore => {
  return configureStore({
    reducer: {
      navigation: navigationReducer,
      quickHelp: quickHelpReducer,
    },
    preloadedState,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        immutableCheck: { warnAfter: 256 },
        serializableCheck: { warnAfter: 256 },
      }),
  });
};

export const createTestStore = (preloadedState: DeepPartial<RootState> = {}): EnhancedStore => {
  const state: RootState = {
    // @ts-ignore
    navigation: {
      ...navigationInitialState,
      ...(preloadedState.navigation || {}),
    },
    // @ts-ignore
    quickHelp: {
      ...quickHelpInitialState,
      ...(preloadedState.quickHelp || {}),
    },
  };
  const store: EnhancedStore = _createStore(state);
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
};
