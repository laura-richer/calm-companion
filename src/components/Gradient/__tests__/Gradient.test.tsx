import React from 'react';
import { Text } from 'react-native';
import { screen } from '@testing-library/react-native';
import { renderWithProviders } from '~utils/test-utils';
import GradientDark from '../GradientDark';
import GradientLight from '../GradientLight';

const _renderGradientLight = () => {
  return renderWithProviders(
    <GradientLight>
      <Text>Child content</Text>
    </GradientLight>
  );
};

const _renderGradientDark = () => {
  return renderWithProviders(
    <GradientDark>
      <Text>Child content</Text>
    </GradientDark>
  );
};

describe('gradient', () => {
  test('matches dark snapshot', () => {
    _renderGradientDark();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('matches light snapshot', () => {
    _renderGradientLight();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('renders dark child content', () => {
    _renderGradientDark();
    const content = screen.getByText('Child content');
    expect(content).toBeOnTheScreen();
  });

  test('renders light child content', () => {
    _renderGradientLight();
    const content = screen.getByText('Child content');
    expect(content).toBeOnTheScreen();
  });
});
