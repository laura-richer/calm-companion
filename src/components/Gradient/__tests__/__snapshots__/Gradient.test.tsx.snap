// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gradient matches dark snapshot 1`] = `
<BVLinearGradient
  angle={170}
  angleCenter={
    {
      "x": 0.75,
      "y": 0.75,
    }
  }
  colors={
    [
      4282350466,
      4287349689,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  useAngle={true}
>
  <Text>
    Child content
  </Text>
</BVLinearGradient>
`;

exports[`gradient matches light snapshot 1`] = `
<BVLinearGradient
  angle={145}
  angleCenter={
    {
      "x": 0.75,
      "y": 0.75,
    }
  }
  colors={
    [
      4290041024,
      4292997582,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  useAngle={true}
>
  <Text>
    Child content
  </Text>
</BVLinearGradient>
`;
